prusti-specs/src/export_spec_rewriter.rs

```diff
@@ -0,0 +1,488 @@
+use proc_macro2::{TokenStream, TokenTree};
+use quote::{quote, quote_spanned, ToTokens};
+use syn::{parse::Parse, spanned::Spanned, Token};
+
+/// An abstraction for tokens that user use to specify method trait bound
+/// #[trait_bound(<T: Trait>)] | #[trait_bound(<T>)] | #[trait_bound(<T> where T: Trait1 + Trait2 ...)]
+#[derive(Debug)]
+pub struct PrustiTraitBound(syn::Generics);
+
+impl Parse for PrustiTraitBound {
+    fn parse(input: syn::parse::ParseStream) -> syn::Result<Self> {
+        let mut generics: syn::Generics = input.parse()?;
+        let colon_token: Option<Token![:]> = input.parse()?;
+        let mut bounds = syn::punctuated::Punctuated::new();
+        if colon_token.is_some() {
+            loop {
+                if input.peek(Token![where]) || input.peek(Token![=]) || input.peek(Token![;]) {
+                    break;
+                }
+                bounds.push_value(input.parse::<syn::TypeParamBound>()?);
+                if input.peek(Token![where]) || input.peek(Token![=]) || input.peek(Token![;]) {
+                    break;
+                }
+                bounds.push_punct(input.parse::<Token![+]>()?);
+            }
+        }
+        generics.where_clause = input.parse()?;
+        Ok(Self(generics))
```

Comment:
Can this whole thing not just be a `parse` with a `Punctuated<WherePredicate, Token![,]>` result?

Reply:
Yeah I think this is related to what kind of trait bound syntax we want to support for the attribute. I was trying to support all valid syntax for impl with generics bounds (this is a little bit overkill)